client/src/components/History/Content/model/StatesInfo.vue

```diff
@@ -0,0 +1,65 @@
+<script setup lang="ts">
+import { STATES } from "./states";
+import type { ComputedRef } from "vue";
+import { computed } from "vue";
+
+type State = {
+    status: string;
+    text?: string;
+    icon?: string;
+    spin?: boolean;
+};
+interface States {
+    [key: string]: State;
+}
+interface HelpText {
+    [key: string]: string;
+}
+
+const props = defineProps({
+    showHelp: { type: Boolean, default: false },
+});
+
+const emit = defineEmits<{
+    (e: "update:show-help", showHelp: boolean): void;
+    (e: "set-filter", filter: string, value: string): void;
+}>();
+
+const propShowHelp = computed({
+    get: () => {
+        return props.showHelp;
+    },
+    set: (val) => {
+        emit("update:show-help", val);
+    },
+});
+const states: ComputedRef<States> = computed(() => STATES);
+
+const helpText: HelpText = {
+    failed: "The dataset failed to run.",
+    ok: "The dataset has no errors and is useable and readable in the history.",
+};
+
+function onFilter(value: string) {
+    propShowHelp.value = false;
+    emit("set-filter", `state:`, value);
+}
+</script>
+
+<template>
```

Comment:
That's really cool, in a follow up we couldalso list the dataset state in the info page, together with the StatesInfo component:
<img width="1154" alt="Screenshot 2023-03-15 at 18 15 51" src="https://user-images.githubusercontent.com/6804901/225389002-07a925dd-545c-4048-8dda-27241cbb6dcb.png">

Reply:
yes, that's a nice idea